test/topology_custom/test_truncate_with_tablets.py

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from cassandra.query import SimpleStatement, ConsistencyLevel
+from test.pylib.manager_client import ManagerClient
+from test.topology.conftest import skip_mode
+import pytest
+import logging
+import asyncio
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_truncate_while_migration(manager: ManagerClient):
+
+    logger.info('Bootstrapping cluster')
+    cfg = { 'enable_tablets': True,
+            'error_injections_at_startup': ['migration_streaming_wait']
+            }
+
+    servers = []
+    servers.append(await manager.server_add(config=cfg))
+
+    cql = manager.get_cql()
+
+    # Create a keyspace with tablets and initial_tablets == 2, then insert data
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 2}")
+    await cql.run_async('CREATE TABLE test.test (pk int PRIMARY KEY, c int);')
+
+    keys = range(1024)
+    await asyncio.gather(*[cql.run_async(f'INSERT INTO test.test (pk, c) VALUES ({k}, {k});') for k in keys])
+
+    # Add a node to the cluster. This will cause the tablet load balancer to migrate one tablet to the new node
+    servers.append(await manager.server_add(config=cfg))
+
+    # Wait for tablet streaming to start
+    pending_node = servers[1]
+    pending_log = await manager.server_open_log(pending_node.server_id)
+
+    await pending_log.wait_for('migration_streaming_wait: start')
+    await manager.api.message_injection(pending_node.ip_addr, 'migration_streaming_wait')
+
+    # Do a TRUNCATE TABLE while the tablet is being streamed
+    await cql.run_async(SimpleStatement('TRUNCATE TABLE test.test'))
+
+    # Wait for streaming to complete
```

Comment:
How long does the test take to run?

Do you not need to elevate the timeouts?

Reply:
This is on my laptop:

```bash-5.2$ time ./test.py topology_custom/test_truncate_with_tablets
Found 1 tests.
Starting S3 mock server on ('127.73.232.1', 2012)
================================================================================
[N/TOTAL]   SUITE    MODE   RESULT   TEST
------------------------------------------------------------------------------
[1/1]      topology_custom   dev   [ PASS ] topology_custom.test_truncate_with_tablets.1
Stopping S3 mock server
------------------------------------------------------------------------------
CPU utilization: 2.6%

real	0m6.953s
user	0m2.262s
sys	0m1.082s
```
The default timeout for truncate is 30 seconds.